test/test_dynamic_shapes.py

```diff
@@ -1770,6 +1772,36 @@ def test_dim_constraints_solve_full(self):
             "dynamic_dim(L['x9'], 1) == dynamic_dim(L['x6'], 1)",
         })
 
+        def dummy_f(x0, x1, x2, x3, x4, x5, x6, x7, x8, x9, x11, x12):
+            pass
+
+        action_code = dim_constraints.prettify_results(inspect.signature(dummy_f))
+        static_code, dynamic_code = re.findall(r"```(.*?)```", action_code, re.DOTALL)
+        expected_static = '''
+def static_constraints(x0, x1, x2, x3, x4, x5, x6, x7, x8, x9, x11, x12):
+    x0.size()[0] = 8
+    x1.size()[2] = 96
+    x11.size()[1] == 1
+    x12.size()[2] == 3
+    x2.size()[0] == 8
+    x3.size()[0] == 8
+    x4.size()[0] == 8
+    x5.size()[1] = 22
+    x7.size()[3] == 96
+    x8.size()[1] == 22
```

Comment:
We expect users to execute as the x0-x1 in exported program may be different in user program e.g. it could be 
```python
args = [t1, t2, t3]
def f(x0, x1, x2):
  # computations on x0, x1, x2
  pass
torchdynamo.export(f, *args)
```
In this case, directly returned constraints may look like this:
```python
constraints = [
        2 <= dynamic_dim(x1, 1),
        dynamic_dim(x2, 1) == dynamic_dim(x1, 1),
        dynamic_dim(x1, 1) == dynamic_dim(x3, 1),
]
```
It's not directly copy-pastable: users need to find the mapping between t1, t2, t3 and x0, x1, x2. If we export a function with the same signature, we could directly call specify_constriants like below:
```python
def specify_constraints(x0, x2, x2):
  return [
        2 <= dynamic_dim(x1, 1),
        dynamic_dim(x2, 1) == dynamic_dim(x1, 1),
        dynamic_dim(x1, 1) == dynamic_dim(x3, 1),
]
constraints = specify_constraitns(*args)
```
Does it make sense?

Reply:
I guess I am curious why the static constraints like `x7.size()[3] == 96` are wrapped inside function that doesn't return anything. For dynamic constraints, it totally makes sense :)

Reply:
Ohh, sorry I thought you were referring to static ones.  For the static constraints, we'll call it specializations and make it return a boolean in this revision. It seems to be prettier than the L[""] and more consistent with specify_constraints.